libcloudforensics/providers/gcp/internal/storage.py

```diff
@@ -83,10 +85,65 @@ def GetObjectMetadata(self,
       Dict: An API operation object for a Google Cloud Storage object.
            https://cloud.google.com/storage/docs/json_api/v1/objects#resource
     """
-
+    if not gcs_path.startswith('gs://'):
+      gcs_path = 'gs://' + gcs_path
     bucket, object_path = SplitGcsPath(gcs_path)
     gcs_api_client = self.GcsApi().objects()
     request = gcs_api_client.get(
         bucket=bucket, object=object_path, userProject=user_project)
     response = request.execute()  # type: Dict[str, Any]
     return response
+
+  def GetBucketACLs(self,
+                    bucket: str,
+                    user_project: Optional[str] = None) -> Dict[str, List[str]]:
+    """Get ACLs for a Google Cloud Storage bucket.
+
+    This includes both ACL entries and IAM policies.
+
+    Args:
+      bucket (str): Name of a bucket in GCS.
+          Ex: logs_bucket_1
+      user_project (str): The project ID to be billed for this request.
+          Required for Requester Pays buckets.
+
+    Returns:
+      Dict: A mapping of role to members of that role.
+    """
+    ret = collections.defaultdict(list)
+    if bucket.startswith('gs://'):
+      # Can change to removeprefix() in 3.9
+      bucket = bucket[5:]
+    gcs_bac = self.GcsApi().bucketAccessControls()
+    request = gcs_bac.list(
+        bucket=bucket, userProject=user_project)
+    # https://cloud.google.com/storage/docs/json_api/v1/bucketAccessControls#resource
+    ac_response = request.execute()
+    for item in ac_response.get('items', []):
+      if item.get('kind') == 'storage#bucketAccessControl':  # Sanity check
+        ret[item['role']].append(item['entity'])
+    gcs_buckets = self.GcsApi().buckets()
+    request = gcs_buckets.getIamPolicy(bucket=bucket)
+    # https://cloud.google.com/storage/docs/json_api/v1/buckets/getIamPolicy
+    iam_response = request.execute()
+    for item in iam_response.get('bindings', []):
+      for member in item.get('members', []):
+        ret[item['role']].append(member)
+    return ret
+
+  def ListBucketObjects(self, bucket: str) -> List[Dict[str, Any]]:
+    """List objects (with metadata) in a Google Cloud Storage bucket.
+
+    Args:
+      bucket (str):  Name of a bucket in GCS.
+
+    Returns:
+      List of Object Dicts (see GetObjectMetadata)
+    """
+    if bucket.startswith('gs://'):
+      # Can change to removeprefix() in 3.9
+      bucket = bucket[5:]
+    gcs_api_client = self.GcsApi().objects()
```

Comment:
I find it a little convoluted that we have a class variable `self.gcs_api_client` which the function `GcsApi()` set and returns, and later we use the same name as a local variable but now we're calling `.objects()` on it. Perhaps something worth refactoring?

Reply:
Agreed, and changed.